src/Agent.Listener/JobDispatcher.cs

```diff
@@ -466,7 +466,19 @@ private async Task RunAsync(Pipelines.AgentJobRequestMessage message, WorkerDisp
                     // send notification to machine provisioner.
                     var systemConnection = message.Resources.Endpoints.SingleOrDefault(x => string.Equals(x.Name, WellKnownServiceEndpointNames.SystemVssConnection, StringComparison.OrdinalIgnoreCase));
                     var accessToken = systemConnection?.Authorization?.Parameters["AccessToken"];
-                    await notification.JobStarted(message.JobId, accessToken, systemConnection.Url);
+                    VariableValue buildId = null;
+                    VariableValue definitionId = null;
+                    if (message.Plan.PlanType == "Build")
+                    {
+                        message.Variables.TryGetValue("build.buildId", out buildId);
+                        message.Variables.TryGetValue("system.definitionId", out definitionId);
+                    }
+                    else if (message.Plan.PlanType == "Release")
+                    {
+                        message.Variables.TryGetValue("release.deploymentId", out buildId);
+                        message.Variables.TryGetValue("release.definitionId", out definitionId);
+                    }
+                    await notification.JobStarted(message.JobId, accessToken, systemConnection.Url, message.Plan.PlanId, buildId.Value, definitionId.Value, message.Plan.PlanType);
```

Comment:
Again, i'm hoping the process monitor runs with both versions of the agent. One with less parameters and then this.

Reply:
Yes processmonitor have changes to support both the versions, we have raised the PR but it depends on MMS end that they deploy on all pools. Currently they have deployed fr windows and linux, in following rings 0,1 and 2.